src/main/scala/is/hail/utils/richUtils/RichCodeMemoryBuffer.scala

```diff
@@ -79,6 +80,36 @@ class RichCodeMemoryBuffer(val region: Code[MemoryBuffer]) extends AnyVal {
     region.invoke[Long, Long, Boolean]("loadBit", byteOff, bitOff)
   }
 
+  def loadAnnotation(typ: Type): Code[Long] => Code[_] = typ match {
+    case TBoolean =>
+      this.loadBoolean(_)
+    case TInt32 =>
+      this.loadInt(_)
+    case TInt64 =>
+      this.loadLong(_)
+    case TFloat32 =>
+      this.loadFloat(_)
+    case TFloat64 =>
+      this.loadDouble(_)
+    case _ =>
+      off => off
+  }
+
+  def appendAnnotation(typ: Type): (Code[_]) => Code[_] = typ match {
```

Comment:
Why is the return type not `(Code[_]) => Code[Unit]`?

Reply:
Mostly because of bad experiences I had with Scala implicitly converting lots of expressions to be of type `Unit`, whereas it doesn't do that if the type annotation is `_` (i.e. existential). I think you're right that in this case there's no danger of that because I always work opaquely with `Code[T]` never with `T`s.